Infrastructure: move away from constructor initialisation lists (part 3)

SlySven opened the pull request:
Start to process the outstanding items in #4578:
* Some clean up of the AnnoucerXxxx files - insert missing `pre_guard` and `post_guard` header files.
* `EAction` - improve the constructor to take two more arguments that the class needs. By doing so it is possible to make the two class members that retain those values `private` - eliminating a `TODO` comment!
* `dlgMapLabel` - move a couple of (non-standard) colour initialisations to the header file. Also remove a dummy destructor. Convert a `QStringLiteral(...)` wrapper to our `qsl(...)` one.
* `dlgModuleManager` - switch from the "Direct" to the "Multiple Inheritance" approach. This means we can remove a number of pointers and refer to members from the form/ dialogue directly in the class by their direct identifier. Insert missing `pre_guard`/`post_guard` Mudlet headers to `.cpp` file. Add missing `override` to the destructor.
* `dlgPackageManager` - same changes as per `dlgModuleManger`.

For the differences between the "Direct" and "Multiple Inheritance" approaches mentioned above see:
https://doc.qt.io/qt-5/designer-using-a-ui-file.html

Unfortunately, the `dlgPackageExporter` class is NOT amenable to this simplification as it uses a custom widget (derived from `QTextEdit`) whose class is also defined in the same file.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Infrastructure: move away from constructor initialisation lists (part 3)

Start to process the outstanding items in #4578:
* Some clean up of the AnnoucerXxxx files - insert
missing `pre_guard` and `post_guard` header files.
* `EAction` - improve the constructor to take two more
arguments that the class needs. By doing so it is
possible to make the two class members that retain those
values `private` - eliminating a `TODO` comment!
* `dlgMapLabel` - move a couple of (non-standard) colour
initialisations to the header file. Also remove a dummy
destructor. Convert a `QStringLiteral(...)` wrapper to
our `qsl(...)` one.
* `dlgModuleManager` - switch from the "Direct" to the
"Multiple Inheritance" approach. This means we can remove
a number of pointers and refer to members from the form/
dialogue directly in the class by their direct
identifier. Insert missing `pre_guard`/`post_guard`
Mudlet headers to `.cpp` file. Add missing `override` to
the destructor.
* `dlgPackageManager` - same changes as per
`dlgModuleManger`.

For the differences between the "Direct" and "Multiple
Inheritance" approaches mentioned above see:
https://doc.qt.io/qt-5/designer-using-a-ui-file.html

Unfortunately, the `dlgPackageExporter` class is NOT
amenable to this simplification as it uses a custom
widget (derived from `QTextEdit`) whose class is also
defined in the same file.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/dlgModuleManager.cpp:
```diff
@@ -1,7 +1,7 @@
  *   Copyright (C) 2011 by Chris Mitchell                                  *
- *   Copyright (C) 2021 by Stephen Lyons - slysven@virginmedia..com        *
+ *   Copyright (C) 2021 by Manuel Wegmann - wegmann.manuel@yahoo.com       *
+ *   Copyright (C) 2021-2022 by Stephen Lyons - slysven@virginmedia..com   *
```
```suggestion
 *   Copyright (C) 2021-2022 by Stephen Lyons - slysven@virginmedia.com    *
```

SlySven committed:
Update src/dlgModuleManager.cpp

Fix typo in Copyright boilerplate comment.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

Co-authored-by: Vadim Peretokin <vperetokin@hey.com>

SlySven committed:
Merge branch 'development' into Infrastructure_moveAwayFromConstructorInitialisationLists_part3
